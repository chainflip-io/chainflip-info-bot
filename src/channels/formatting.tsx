export const Bold = ({
  children,
  channel,
}: {
  children: React.ReactNode;
  channel: 'discord' | 'telegram';
}) => (channel === 'discord' ? <>**{children}**</> : <strong>{children}</strong>);
