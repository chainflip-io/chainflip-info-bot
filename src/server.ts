import { createBullBoard } from '@bull-board/api';
import { BullMQAdapter } from '@bull-board/api/bullMQAdapter.js';
import { FastifyAdapter } from '@bull-board/fastify';
import { type Queue } from 'bullmq';
import fastify from 'fastify';
import { GraphQLClient } from 'graphql-request';
import env from './env.js';
import logger from './utils/logger.js';

const app = fastify({ loggerInstance: logger, disableRequestLogging: true });

export const explorerClient = new GraphQLClient(env.EXPLORER_GATEWAY_URL);
export const lpClient = new GraphQLClient(env.LP_GATEWAY_URL);

export const createServer = (queues: Queue[]) => {
  const serverAdapter = new FastifyAdapter();

  createBullBoard({
    queues: queues.map((q) => new BullMQAdapter(q)),
    serverAdapter,
  });

  const basePath = '/admin/queues';
  serverAdapter.setBasePath(basePath);
  app.register(serverAdapter.registerPlugin(), { prefix: basePath, basePath: '/' });

  app.get('/health', () => ({ status: 'ok' }));

  app.get('/', (req, res) => res.redirect('/admin/queues', 302));

  return app;
};
