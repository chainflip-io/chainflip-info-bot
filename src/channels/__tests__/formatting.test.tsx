import { renderToStaticMarkup } from 'react-dom/server';
import { describe, expect, it } from 'vitest';
import { Bold } from '../formatting.js';

describe('Bold', () => {
  it.each(['telegram', 'discord'] as const)('formats %s messages correctly', (channel) => {
    expect(renderToStaticMarkup(<Bold channel={channel}>hello</Bold>)).toMatchSnapshot(channel);
  });
});
