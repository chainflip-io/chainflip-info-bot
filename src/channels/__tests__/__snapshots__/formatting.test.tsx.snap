// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bold > formats discord messages correctly > discord 1`] = `"**hello**"`;

exports[`Bold > formats telegram messages correctly > telegram 1`] = `"<strong>hello</strong>"`;
